Monika After Story/game/script-topics.rpy

```diff
@@ -12905,3 +12905,205 @@ label monika_enjoyingspring:
     else:
         m 2rkc "...but I guess there's no real way to avoid it, is there?"
     return
+
+# start of bookmark and derandom related events
+default persistent._mas_player_bookmarked = {}
+# dict to store bookmarked events
+default persistent._mas_player_derandomed = {}
+# dict to store player derandomed events
+default persistent.flagged_monikatopic = None
+# var set when we flag a topic for derandom
+
+init python:
+    def derandom_topic():
+        """
+        Function for the derandom hotkey, 'x'
+        """
+        curr_topic = persistent.current_monikatopic
+        if (
+            mas_getEV(curr_topic) is not None
+            and mas_getEV(curr_topic).random
+            and curr_topic.startswith("monika_")
+            and mas_getEV(curr_topic).prompt != mas_getEV(curr_topic).eventlabel
+        ):
+            if not "mas_topic_derandom" in persistent.event_list:
+                persistent.flagged_monikatopic = curr_topic
+                pushEvent('mas_topic_derandom',True)
```

Comment:
something to note, but pushing events during a topic should in already skip the loop evaluation. although i think i would leave this as is just to make sure.